askama_shared/src/escaping.rs

```diff
@@ -55,26 +55,33 @@ fn escapable(b: u8) -> bool {
 }
 
 pub fn escape(s: String) -> String {
-    let mut found = Vec::new();
-    for (i, b) in s.as_bytes().iter().enumerate() {
-        if escapable(*b) {
-            found.push(i);
+    let found: usize = if let Some(c) = s.chars().next() {
+        if escapable(c as u8) {
+            0
+        } else {
+            let mut found = 0;
+            for (i, b) in s.as_bytes()[1..].iter().enumerate() {
+                if escapable(*b) {
+                    found = i + 1;
+                    break;
+                }
+            }
+
+            if found == 0 {
+                return s;
+            }
+
+            found
         }
-    }
-    if found.is_empty() {
+    } else {
         return s;
-    }
+    };
 
     let bytes = s.as_bytes();
-    let max_len = bytes.len() + found.len() * 5;
-    let mut res = Vec::<u8>::with_capacity(max_len);
-    let mut start = 0;
-    for idx in &found {
-        if start < *idx {
-            res.extend(&bytes[start..*idx]);
-        }
-        start = *idx + 1;
-        match bytes[*idx] {
+    let mut res = Vec::with_capacity(s.len() + 5);
+    res.extend(&bytes[0..found]);
+    for c in bytes[found..].iter() {
```

Comment:
I'm also skeptical of this transformation for longer string sizes. Let's land a more comprehensive set of benchmark cases before you try to optimize, ok?

Reply:
The rest I look at it with a longer text as you indicate above. Now I upload the changes.